modules/core/src/webgl/classes/texture-cube.js

```diff
@@ -81,15 +84,30 @@ export default class TextureCube extends Texture {
     const {gl} = this;
     const imageDataMap = pixels || data;
 
-    return Promise.all(FACES.map(face => imageDataMap[face])).then(resolvedFaces => {
+    return Promise.all(
+      FACES.map(face => {
+        const facePixels = imageDataMap[face];
+        return Promise.all(Array.isArray(facePixels) ? facePixels : [facePixels]);
+      })
+    ).then(resolvedFaces => {
       this.bind();
 
       FACES.forEach((face, index) => {
-        if (width && height) {
-          gl.texImage2D(face, 0, format, width, height, border, format, type, resolvedFaces[index]);
-        } else {
-          gl.texImage2D(face, 0, format, format, type, resolvedFaces[index]);
+        if (resolvedFaces[index].length > 1 && this.opts.mipmaps !== false) {
+          log.warn(
+            `Texture ${
```

Comment:
We normally avoid long error messages that add to the bundle size but are very seldom used.

Add as comment and log something minimal, user can click their way back here and read the comment.

Reply:
Ok